forgerock-core/src/main/java/org/forgerock/android/auth/EncryptedSharedPreferences.kt

```diff
@@ -67,22 +83,45 @@ class EncryptedPreferences {
         }
 
         // Creates the instance for the encrypted preferences.
-        private fun createPreferencesFile(context: Context,
-                                              fileName: String,
-                                              aliasName: String): SharedPreferences {
-
-
-            val masterKeyAlias = MasterKey.Builder(context, aliasName)
-                .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
-                .build()
-
-            return EncryptedSharedPreferences.create(
+        private fun createPreferencesFile(
+            context: Context,
+            fileName: String,
+            aliasName: String? = null
+        ): SharedPreferences {
+            val builder =
+                aliasName?.let { MasterKey.Builder(context, it) } ?: MasterKey.Builder(context)
+            builder.setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
+            return  EncryptedSharedPreferences.create(
                 context,
                 fileName,
-                masterKeyAlias,
+                builder.build(),
                 EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
                 EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
             )
         }
+
+        // master key recovery code
+        private fun recoverFromDefaultMasterKey(
```

Comment:
May provide javadoc, that this method is to extract data which encrypted with MasterKey, and delete the master key.